app/request_handler.py

```diff
@@ -184,31 +184,35 @@ def _guessYelpId(placeName, lat, lon):
         return cachedId
 
     opts = {
-      # 'term': placeName,
+      'term': placeName,
       'limit': 20,
-      'radius_filter': 1000,
+      'radius_filter': 100,
       #'radius_filter': 100,
       #'sort_by': 'distance',
       'sort': 1,
     }
     r = yelpClient.search_by_coordinates(lat, lon, **opts)
-    if len(r.businesses) > 0:
-        location = (lat, lon)
-        biz = min(r.businesses, key=lambda b: 
-            geo.distance(location, 
-                         (b.location.coordinate.latitude, b.location.coordinate.longitude))
-        )
-        log.debug("%s --> %s" % (placeName, biz.name))
-        researchVenue(biz)
-
-        # Add bizId to cache
-        record = { "cache/" +  safePlaceName : str(biz.id) }
-        db.child(eventsTable).update(record)
-
-        return biz.id
-    else:
-        log.info("Can't find %s" % placeName)
-        return None
+    try:
+        if len(r.businesses) > 0:
+            location = (lat, lon)
+            biz = min(r.businesses, key=lambda b:
+                geo.distance(location,
+                             (b.location.coordinate.latitude, b.location.coordinate.longitude))
+            )
+            log.debug("%s --> %s" % (placeName, biz.name))
+            researchVenue(biz)
+
+            # Add bizId to cache
+            record = { "cache/" +  safePlaceName : str(biz.id) }
+            db.child(eventsTable).update(record)
+
+            return biz.id
+        else:
+            log.info("Can't find %s" % placeName)
+            return None
+    except:
+        log.error("Error parsing yelp businesses for " + placeName)
+
```

Comment:
So, if any one of the businesses don't have a location, then this fails. This feels… fragile.

Suggest filtering out the businesses without a location: 

```python
            businessWithCoords = filter(
                lambda b: 
                    (b.location is not None) and (b.location.coordinate is not None),
                r.businesses)
            biz = min(businessWithCoords, key=lambda b:
                geo.distance(location,
                             (b.location.coordinate.latitude, b.location.coordinate.longitude))
            )
```